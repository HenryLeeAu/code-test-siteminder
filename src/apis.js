export const baseApi = 'http://www.omdbapi.com/?&apikey=81f5bea7';
