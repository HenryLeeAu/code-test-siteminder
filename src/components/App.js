import React, { Component } from 'react';
import SearchBox from 'components/SearchBox';
import MovieList from 'components/MovieList';
import MovieDetail from 'components/MovieDetail';
import Pagination from 'components/Pagination';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="sidebar">
          <SearchBox />
          <div className="movieListWrap">
            <MovieList />
          </div>
          <Pagination />
        </div>
        <div className="detail">
          <MovieDetail />
        </div>
      </div>
    );
  }
}

export default App;
