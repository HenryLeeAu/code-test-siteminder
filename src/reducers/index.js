import { combineReducers } from 'redux';
import searchStatusReducer from 'reducers/searchStatus';

export default combineReducers({
  searchStatus: searchStatusReducer,
});
