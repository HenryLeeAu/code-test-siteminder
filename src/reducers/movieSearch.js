import {
  INPUT_KEYWORD,
  GET_LIST,
  SHOW_DETAIL,
} from 'actions/types';

const defaultState = {
  keyword: '',
  currentPage: '',
  totalPage: '',
  totalNum: '',
  fetchMovieList: [],
  currentMovieDetail: null,
  fetchedMovies: [],
};
export default function(state = defaultState, action) {
  switch (action.type) {
    case INPUT_KEYWORD:
      return { ...state, keyword: action.payload };
    case GET_LIST:
      const totalPage = Math.ceil(action.payload.data.totalResults / 10);
      return {
        ...state,
        currentPage: 1,
        totalPage,
        fetchMovieList: action.payload.data.Search,
        totalNum: action.payload.data.totalResults,
      };
    case SHOW_DETAIL:
      return {
        ...state,
        currentMovieDetail: action.payload.data,
        fetchedMovies: [...state.fetchedMovies, action.payload.data],
      };
    default:
      return state;
  }
}
