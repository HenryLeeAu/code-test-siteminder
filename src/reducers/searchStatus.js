import {
  INPUT_KEYWORD,
  GET_LIST,
  SHOW_DETAIL,
  RECOVER_DETAIL,
  UPDATE_CURRENT_PAGE
} from "actions/types";

const defaultState = {
  keyword: "",
  currentPage: null,
  totalPages: "",
  totalNum: "",
  currentId:null,
  currentMovieList: [],
  currentMovieDetail: null,

  visitedMovieDetailList: []
};
export default function(state = defaultState, action) {
  switch (action.type) {
    case INPUT_KEYWORD:
      return { ...state, keyword: action.payload };
    case GET_LIST:
      const totalPages = Math.ceil(action.payload.data.totalResults / 10);
      return {
        ...state,
        currentPage:action.payload.fetchPage,
        totalPages,
        currentMovieList: action.payload.data.Search,
        totalNum: action.payload.data.totalResults
      };
    case SHOW_DETAIL:
      return {
        ...state,
        currentId:action.payload.data.imdbID,
        currentMovieDetail: action.payload.data,
        visitedMovieDetailList: [
          ...state.visitedMovieDetailList,
          action.payload.data
        ]
      };
    case RECOVER_DETAIL:
      return {
        ...state,
        currentMovieDetail: action.payload.matchedMovie
      };
    case UPDATE_CURRENT_PAGE:
      return {
       ...state,
       currentPage:action.payload.updatedNum
     }
   
    default:
      return state;
  }
}
