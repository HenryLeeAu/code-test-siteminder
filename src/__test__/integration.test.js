import React from 'react';
import { mount } from 'enzyme';
import moxios from 'moxios';
import Root from 'Root';
import App from 'components/App';
import { baseApi } from 'apis';
import { defaultState } from 'reducers/searchStatus';
describe('Intergation', () => {
  beforeEach(() => {
    moxios.install();
    moxios.stubRequest(`${baseApi}&s=marvel&page=1`, {
      status: 200,
      response: {
        Search: [
          {
            Title: 'Marvel One-Shot: Item 47',
            Year: '2012',
            imdbID: 'tt2247732',
            Type: 'movie',
            Poster:
              'https://m.media-amazon.com/images/M/MV5BMTM2MzY1ODcyN15BMl5BanBnXkFtZTcwNTE3OTIxOA@@._V1_SX300.jpg',
          },
        ],
        totalResults: '143',
        Response: 'True',
      },
    });
    moxios.stubRequest(`${baseApi}&i=tt2247732`, {
      status: 200,
      Title: 'The Merv Griffin Show',
      response: {
        Title: 'The Merv Griffin Show',
        Year: '1962â€“1986',
        Rated: 'N/A',
        Released: '01 Oct 1962',
        Runtime: '60 min',
        Genre: 'Comedy, Family, Music',
        Director: 'N/A',
        Writer: 'N/A',
        Actors: 'Merv Griffin',
        Plot:
          'Merv Griffin invites a series of actors, actresses, writers, and directors to discuss the progressive work they have done and current culture, arts, and entertainment surrounding the numerous projects.',
        Language: 'English',
        Country: 'USA',
        Awards:
          'Nominated for 1 Golden Globe. Another 11 wins & 19 nominations.',
        Poster:
          'https://images-na.ssl-images-amazon.com/images/M/MV5BMjA2NjcyMjI5N15BMl5BanBnXkFtZTcwODgyNjMzMQ@@._V1_SX300.jpg',
        Ratings: [{ Source: 'Internet Movie Database', Value: '6.7/10' }],
        Metascore: 'N/A',
        imdbRating: '6.7',
        imdbVotes: '302',
        imdbID: 'tt0055691',
        Type: 'series',
        totalSeasons: '18',
        Response: 'True',
      },
    });
  });

  afterEach(() => {
    moxios.uninstall();
  });
  it('keyin to search list', done => {
    const initialState = {
      searchStatus: defaultState,
    };
    const wrappered = mount(
      <Root initialState={initialState}>
        <App />
      </Root>
    );

    wrappered.find('input').simulate('change', {
      target: { value: 'marvel' },
    });

    setTimeout(() => {
      wrappered.update();
      moxios.wait(() => {
        expect(wrappered).toMatchSnapshot();
        done();
        wrappered.unmount();
      });
    }, 1000);
  });

  it('click movie to show detail', done => {
    moxios.install();
    moxios.stubRequest(`${baseApi}&s=marvel&page=1`, {
      status: 200,
      response: {
        Search: [
          {
            Title: 'Marvel One-Shot: Item 47',
            Year: '2012',
            imdbID: 'tt2247732',
            Type: 'movie',
            Poster:
              'https://m.media-amazon.com/images/M/MV5BMTM2MzY1ODcyN15BMl5BanBnXkFtZTcwNTE3OTIxOA@@._V1_SX300.jpg',
          },
        ],
        totalResults: '143',
        Response: 'True',
      },
    });
    const initialState = {
      searchStatus: {
        keyword: 'marvel',
        currentPage: '1',
        totalPages: '1',
        currentId: null,
        currentMovieDetail: null,
        currentMovieList: [
          {
            imdbID: 'tt2247732',
            Title: 'Marvel',
            Year: '2016',
          },
        ],
      },
    };
    const wrappered = mount(
      <Root initialState={initialState}>
        <App />
      </Root>
    );
    expect(wrappered).toMatchSnapshot();

    wrappered
      .find('.movieList')
      .find('li')
      .simulate('click', 'tt2247732');

    moxios.wait(() => {
      wrappered.update();
      expect(wrappered).toMatchSnapshot();
      done();
      wrappered.unmount();
    });
  });
});
