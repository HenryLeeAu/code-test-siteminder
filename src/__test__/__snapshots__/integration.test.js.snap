// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intergation click move to show detail 1`] = `
<_default
  initialState={
    Object {
      "searchStatus": Object {
        "currentId": null,
        "currentMovieDetail": null,
        "currentMovieList": Array [
          Object {
            "Title": "Marvel",
            "Year": "2016",
            "imdbID": "tt2247732",
          },
        ],
        "currentPage": "1",
        "keyword": "marvel",
        "totalPages": "1",
      },
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <App>
      <div
        className="App"
      >
        <div
          className="sidebar"
        >
          <Connect(SearchBox)>
            <SearchBox
              decreaseCurrentPage={[Function]}
              fetchMovieDetail={[Function]}
              fetchMovieList={[Function]}
              getList={[Function]}
              increaseCurrentPage={[Function]}
              inputKeyword={[Function]}
              searchStatus={
                Object {
                  "currentId": null,
                  "currentMovieDetail": null,
                  "currentMovieList": Array [
                    Object {
                      "Title": "Marvel",
                      "Year": "2016",
                      "imdbID": "tt2247732",
                    },
                  ],
                  "currentPage": "1",
                  "keyword": "marvel",
                  "totalPages": "1",
                }
              }
              showMovieDetail={[Function]}
              updateCurrentPageNum={[Function]}
            >
              <div
                className="SearchBox"
              >
                <input
                  onChange={[Function]}
                  placeholder="keyword"
                  type="text"
                  value=""
                />
              </div>
            </SearchBox>
          </Connect(SearchBox)>
          <div
            className="movieListWrap"
          >
            <Connect(MovieList)>
              <MovieList
                fetchMovieDetail={[Function]}
                searchStatus={
                  Object {
                    "currentId": null,
                    "currentMovieDetail": null,
                    "currentMovieList": Array [
                      Object {
                        "Title": "Marvel",
                        "Year": "2016",
                        "imdbID": "tt2247732",
                      },
                    ],
                    "currentPage": "1",
                    "keyword": "marvel",
                    "totalPages": "1",
                  }
                }
              >
                <ul
                  className="movieList"
                >
                  <_default
                    clicked={false}
                    key="tt2247732Marvel"
                    onClick={[Function]}
                    title="Marvel"
                    year="2016"
                  >
                    <li
                      className=""
                      name="1"
                      onClick={[Function]}
                    >
                      <div>
                        Marvel
                      </div>
                      <div
                        className="align-right"
                      >
                        2016
                      </div>
                    </li>
                  </_default>
                </ul>
              </MovieList>
            </Connect(MovieList)>
          </div>
          <Connect(Pagination)>
            <Pagination
              decreaseCurrentPage={[Function]}
              increaseCurrentPage={[Function]}
              searchStatus={
                Object {
                  "currentId": null,
                  "currentMovieDetail": null,
                  "currentMovieList": Array [
                    Object {
                      "Title": "Marvel",
                      "Year": "2016",
                      "imdbID": "tt2247732",
                    },
                  ],
                  "currentPage": "1",
                  "keyword": "marvel",
                  "totalPages": "1",
                }
              }
            />
          </Connect(Pagination)>
        </div>
        <div
          className="detail"
        >
          <Connect(MovieDetail)>
            <MovieDetail
              dispatch={[Function]}
              searchStatus={
                Object {
                  "currentId": null,
                  "currentMovieDetail": null,
                  "currentMovieList": Array [
                    Object {
                      "Title": "Marvel",
                      "Year": "2016",
                      "imdbID": "tt2247732",
                    },
                  ],
                  "currentPage": "1",
                  "keyword": "marvel",
                  "totalPages": "1",
                }
              }
            >
              <div
                className="before-select"
              >
                You haven't selected any movie yet!
              </div>
            </MovieDetail>
          </Connect(MovieDetail)>
        </div>
      </div>
    </App>
  </Provider>
</_default>
`;

exports[`Intergation click move to show detail 2`] = `
<_default
  initialState={
    Object {
      "searchStatus": Object {
        "currentId": null,
        "currentMovieDetail": null,
        "currentMovieList": Array [
          Object {
            "Title": "Marvel",
            "Year": "2016",
            "imdbID": "tt2247732",
          },
        ],
        "currentPage": "1",
        "keyword": "marvel",
        "totalPages": "1",
      },
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <App>
      <div
        className="App"
      >
        <div
          className="sidebar"
        >
          <Connect(SearchBox)>
            <SearchBox
              decreaseCurrentPage={[Function]}
              fetchMovieDetail={[Function]}
              fetchMovieList={[Function]}
              getList={[Function]}
              increaseCurrentPage={[Function]}
              inputKeyword={[Function]}
              searchStatus={
                Object {
                  "currentId": "tt0055691",
                  "currentMovieDetail": Object {
                    "Actors": "Merv Griffin",
                    "Awards": "Nominated for 1 Golden Globe. Another 11 wins & 19 nominations.",
                    "Country": "USA",
                    "Director": "N/A",
                    "Genre": "Comedy, Family, Music",
                    "Language": "English",
                    "Metascore": "N/A",
                    "Plot": "Merv Griffin invites a series of actors, actresses, writers, and directors to discuss the progressive work they have done and current culture, arts, and entertainment surrounding the numerous projects.",
                    "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMjA2NjcyMjI5N15BMl5BanBnXkFtZTcwODgyNjMzMQ@@._V1_SX300.jpg",
                    "Rated": "N/A",
                    "Ratings": Array [
                      Object {
                        "Source": "Internet Movie Database",
                        "Value": "6.7/10",
                      },
                    ],
                    "Released": "01 Oct 1962",
                    "Response": "True",
                    "Runtime": "60 min",
                    "Title": "The Merv Griffin Show",
                    "Type": "series",
                    "Writer": "N/A",
                    "Year": "1962â€“1986",
                    "imdbID": "tt0055691",
                    "imdbRating": "6.7",
                    "imdbVotes": "302",
                    "totalSeasons": "18",
                  },
                  "currentMovieList": Array [
                    Object {
                      "Title": "Marvel",
                      "Year": "2016",
                      "imdbID": "tt2247732",
                    },
                  ],
                  "currentPage": "1",
                  "keyword": "marvel",
                  "totalPages": "1",
                }
              }
              showMovieDetail={[Function]}
              updateCurrentPageNum={[Function]}
            >
              <div
                className="SearchBox"
              >
                <input
                  onChange={[Function]}
                  placeholder="keyword"
                  type="text"
                  value=""
                />
              </div>
            </SearchBox>
          </Connect(SearchBox)>
          <div
            className="movieListWrap"
          >
            <Connect(MovieList)>
              <MovieList
                fetchMovieDetail={[Function]}
                searchStatus={
                  Object {
                    "currentId": "tt0055691",
                    "currentMovieDetail": Object {
                      "Actors": "Merv Griffin",
                      "Awards": "Nominated for 1 Golden Globe. Another 11 wins & 19 nominations.",
                      "Country": "USA",
                      "Director": "N/A",
                      "Genre": "Comedy, Family, Music",
                      "Language": "English",
                      "Metascore": "N/A",
                      "Plot": "Merv Griffin invites a series of actors, actresses, writers, and directors to discuss the progressive work they have done and current culture, arts, and entertainment surrounding the numerous projects.",
                      "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMjA2NjcyMjI5N15BMl5BanBnXkFtZTcwODgyNjMzMQ@@._V1_SX300.jpg",
                      "Rated": "N/A",
                      "Ratings": Array [
                        Object {
                          "Source": "Internet Movie Database",
                          "Value": "6.7/10",
                        },
                      ],
                      "Released": "01 Oct 1962",
                      "Response": "True",
                      "Runtime": "60 min",
                      "Title": "The Merv Griffin Show",
                      "Type": "series",
                      "Writer": "N/A",
                      "Year": "1962â€“1986",
                      "imdbID": "tt0055691",
                      "imdbRating": "6.7",
                      "imdbVotes": "302",
                      "totalSeasons": "18",
                    },
                    "currentMovieList": Array [
                      Object {
                        "Title": "Marvel",
                        "Year": "2016",
                        "imdbID": "tt2247732",
                      },
                    ],
                    "currentPage": "1",
                    "keyword": "marvel",
                    "totalPages": "1",
                  }
                }
              >
                <ul
                  className="movieList"
                >
                  <_default
                    clicked={false}
                    key="tt2247732Marvel"
                    onClick={[Function]}
                    title="Marvel"
                    year="2016"
                  >
                    <li
                      className=""
                      name="1"
                      onClick={[Function]}
                    >
                      <div>
                        Marvel
                      </div>
                      <div
                        className="align-right"
                      >
                        2016
                      </div>
                    </li>
                  </_default>
                </ul>
              </MovieList>
            </Connect(MovieList)>
          </div>
          <Connect(Pagination)>
            <Pagination
              decreaseCurrentPage={[Function]}
              increaseCurrentPage={[Function]}
              searchStatus={
                Object {
                  "currentId": "tt0055691",
                  "currentMovieDetail": Object {
                    "Actors": "Merv Griffin",
                    "Awards": "Nominated for 1 Golden Globe. Another 11 wins & 19 nominations.",
                    "Country": "USA",
                    "Director": "N/A",
                    "Genre": "Comedy, Family, Music",
                    "Language": "English",
                    "Metascore": "N/A",
                    "Plot": "Merv Griffin invites a series of actors, actresses, writers, and directors to discuss the progressive work they have done and current culture, arts, and entertainment surrounding the numerous projects.",
                    "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMjA2NjcyMjI5N15BMl5BanBnXkFtZTcwODgyNjMzMQ@@._V1_SX300.jpg",
                    "Rated": "N/A",
                    "Ratings": Array [
                      Object {
                        "Source": "Internet Movie Database",
                        "Value": "6.7/10",
                      },
                    ],
                    "Released": "01 Oct 1962",
                    "Response": "True",
                    "Runtime": "60 min",
                    "Title": "The Merv Griffin Show",
                    "Type": "series",
                    "Writer": "N/A",
                    "Year": "1962â€“1986",
                    "imdbID": "tt0055691",
                    "imdbRating": "6.7",
                    "imdbVotes": "302",
                    "totalSeasons": "18",
                  },
                  "currentMovieList": Array [
                    Object {
                      "Title": "Marvel",
                      "Year": "2016",
                      "imdbID": "tt2247732",
                    },
                  ],
                  "currentPage": "1",
                  "keyword": "marvel",
                  "totalPages": "1",
                }
              }
            />
          </Connect(Pagination)>
        </div>
        <div
          className="detail"
        >
          <Connect(MovieDetail)>
            <MovieDetail
              dispatch={[Function]}
              searchStatus={
                Object {
                  "currentId": "tt0055691",
                  "currentMovieDetail": Object {
                    "Actors": "Merv Griffin",
                    "Awards": "Nominated for 1 Golden Globe. Another 11 wins & 19 nominations.",
                    "Country": "USA",
                    "Director": "N/A",
                    "Genre": "Comedy, Family, Music",
                    "Language": "English",
                    "Metascore": "N/A",
                    "Plot": "Merv Griffin invites a series of actors, actresses, writers, and directors to discuss the progressive work they have done and current culture, arts, and entertainment surrounding the numerous projects.",
                    "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMjA2NjcyMjI5N15BMl5BanBnXkFtZTcwODgyNjMzMQ@@._V1_SX300.jpg",
                    "Rated": "N/A",
                    "Ratings": Array [
                      Object {
                        "Source": "Internet Movie Database",
                        "Value": "6.7/10",
                      },
                    ],
                    "Released": "01 Oct 1962",
                    "Response": "True",
                    "Runtime": "60 min",
                    "Title": "The Merv Griffin Show",
                    "Type": "series",
                    "Writer": "N/A",
                    "Year": "1962â€“1986",
                    "imdbID": "tt0055691",
                    "imdbRating": "6.7",
                    "imdbVotes": "302",
                    "totalSeasons": "18",
                  },
                  "currentMovieList": Array [
                    Object {
                      "Title": "Marvel",
                      "Year": "2016",
                      "imdbID": "tt2247732",
                    },
                  ],
                  "currentPage": "1",
                  "keyword": "marvel",
                  "totalPages": "1",
                }
              }
            >
              <div>
                <div
                  className="detailLeft"
                >
                  <h2
                    className="movie-title"
                  >
                    The Merv Griffin Show
                  </h2>
                  <div
                    className="movie-year"
                  >
                    1962â€“1986
                  </div>
                  <p
                    className="movie-desc"
                  >
                    Merv Griffin invites a series of actors, actresses, writers, and directors to discuss the progressive work they have done and current culture, arts, and entertainment surrounding the numerous projects.
                  </p>
                  <div>
                    genere: 
                    Comedy, Family, Music
                  </div>
                  <p>
                    movie description
                  </p>
                  <div>
                    language: 
                    English
                  </div>
                  <div>
                    Director: 
                    N/A
                  </div>
                  <div>
                    Actor(s): 
                    Merv Griffin
                  </div>
                  <div>
                    Duration : 
                    60 min
                  </div>
                </div>
                <img
                  alt="The Merv Griffin Show"
                  src="https://images-na.ssl-images-amazon.com/images/M/MV5BMjA2NjcyMjI5N15BMl5BanBnXkFtZTcwODgyNjMzMQ@@._V1_SX300.jpg"
                />
              </div>
            </MovieDetail>
          </Connect(MovieDetail)>
        </div>
      </div>
    </App>
  </Provider>
</_default>
`;

exports[`Intergation keyin to search list 1`] = `
<_default
  initialState={
    Object {
      "searchStatus": Object {
        "currentId": null,
        "currentMovieDetail": null,
        "currentMovieList": Array [],
        "currentPage": null,
        "keyword": "",
        "totalNum": null,
        "totalPages": null,
      },
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <App>
      <div
        className="App"
      >
        <div
          className="sidebar"
        >
          <Connect(SearchBox)>
            <SearchBox
              decreaseCurrentPage={[Function]}
              fetchMovieDetail={[Function]}
              fetchMovieList={[Function]}
              getList={[Function]}
              increaseCurrentPage={[Function]}
              inputKeyword={[Function]}
              searchStatus={
                Object {
                  "currentId": null,
                  "currentMovieDetail": null,
                  "currentMovieList": Array [
                    Object {
                      "Poster": "https://m.media-amazon.com/images/M/MV5BMTM2MzY1ODcyN15BMl5BanBnXkFtZTcwNTE3OTIxOA@@._V1_SX300.jpg",
                      "Title": "Marvel One-Shot: Item 47",
                      "Type": "movie",
                      "Year": "2012",
                      "imdbID": "tt2247732",
                    },
                  ],
                  "currentPage": 1,
                  "keyword": "marvel",
                  "totalNum": "143",
                  "totalPages": 15,
                }
              }
              showMovieDetail={[Function]}
              updateCurrentPageNum={[Function]}
            >
              <div
                className="SearchBox"
              >
                <input
                  onChange={[Function]}
                  placeholder="keyword"
                  type="text"
                  value="marvel"
                />
              </div>
            </SearchBox>
          </Connect(SearchBox)>
          <div
            className="movieListWrap"
          >
            <Connect(MovieList)>
              <MovieList
                fetchMovieDetail={[Function]}
                searchStatus={
                  Object {
                    "currentId": null,
                    "currentMovieDetail": null,
                    "currentMovieList": Array [
                      Object {
                        "Poster": "https://m.media-amazon.com/images/M/MV5BMTM2MzY1ODcyN15BMl5BanBnXkFtZTcwNTE3OTIxOA@@._V1_SX300.jpg",
                        "Title": "Marvel One-Shot: Item 47",
                        "Type": "movie",
                        "Year": "2012",
                        "imdbID": "tt2247732",
                      },
                    ],
                    "currentPage": 1,
                    "keyword": "marvel",
                    "totalNum": "143",
                    "totalPages": 15,
                  }
                }
              >
                <ul
                  className="movieList"
                >
                  <_default
                    clicked={false}
                    key="tt2247732Marvel One-Shot: Item 47"
                    onClick={[Function]}
                    title="Marvel One-Shot: Item 47"
                    year="2012"
                  >
                    <li
                      className=""
                      name="1"
                      onClick={[Function]}
                    >
                      <div>
                        Marvel One-Shot: Item 47
                      </div>
                      <div
                        className="align-right"
                      >
                        2012
                      </div>
                    </li>
                  </_default>
                </ul>
              </MovieList>
            </Connect(MovieList)>
          </div>
          <Connect(Pagination)>
            <Pagination
              decreaseCurrentPage={[Function]}
              increaseCurrentPage={[Function]}
              searchStatus={
                Object {
                  "currentId": null,
                  "currentMovieDetail": null,
                  "currentMovieList": Array [
                    Object {
                      "Poster": "https://m.media-amazon.com/images/M/MV5BMTM2MzY1ODcyN15BMl5BanBnXkFtZTcwNTE3OTIxOA@@._V1_SX300.jpg",
                      "Title": "Marvel One-Shot: Item 47",
                      "Type": "movie",
                      "Year": "2012",
                      "imdbID": "tt2247732",
                    },
                  ],
                  "currentPage": 1,
                  "keyword": "marvel",
                  "totalNum": "143",
                  "totalPages": 15,
                }
              }
            >
              <div
                className="pagination"
              >
                <div
                  className="left"
                />
                <div
                  className="middle"
                >
                  <div>
                    1
                     / 
                    15
                     pages
                  </div>
                  143
                   total
                </div>
                <div
                  className="right"
                >
                  <button
                    onClick={[Function]}
                  >
                    next
                  </button>
                </div>
              </div>
            </Pagination>
          </Connect(Pagination)>
        </div>
        <div
          className="detail"
        >
          <Connect(MovieDetail)>
            <MovieDetail
              dispatch={[Function]}
              searchStatus={
                Object {
                  "currentId": null,
                  "currentMovieDetail": null,
                  "currentMovieList": Array [
                    Object {
                      "Poster": "https://m.media-amazon.com/images/M/MV5BMTM2MzY1ODcyN15BMl5BanBnXkFtZTcwNTE3OTIxOA@@._V1_SX300.jpg",
                      "Title": "Marvel One-Shot: Item 47",
                      "Type": "movie",
                      "Year": "2012",
                      "imdbID": "tt2247732",
                    },
                  ],
                  "currentPage": 1,
                  "keyword": "marvel",
                  "totalNum": "143",
                  "totalPages": 15,
                }
              }
            >
              <div
                className="before-select"
              >
                You haven't selected any movie yet!
              </div>
            </MovieDetail>
          </Connect(MovieDetail)>
        </div>
      </div>
    </App>
  </Provider>
</_default>
`;
